frontends/election-manager/src/components/smartcard_modal/card_details_view.tsx

```diff
@@ -56,20 +56,36 @@ export function CardDetailsView({
     actionStatus?.action === 'Program' && actionStatus?.status === 'Success';
 
   async function resetCardPin() {
-    if (!electionDefinition) {
-      return;
-    }
+    assert(electionDefinition);
+    assert(role === 'superadmin' || role === 'admin');
+
     setActionStatus({
       action: 'PinReset',
       role,
       status: 'InProgress',
     });
-    const result = await card.programUser({
-      role: 'admin',
-      electionData: electionDefinition.electionData,
-      electionHash: electionDefinition.electionHash,
-      passcode: generatePin(),
-    });
+    let result;
+    switch (role) {
+      case 'superadmin': {
+        result = await card.programUser({
+          role: 'superadmin',
+          passcode: generatePin(),
+        });
+        break;
+      }
+      case 'admin': {
+        result = await card.programUser({
+          role: 'admin',
+          electionData: electionDefinition.electionData,
+          electionHash: electionDefinition.electionHash,
+          passcode: generatePin(),
+        });
+        break;
+      }
+      default: {
```

Comment:
I think generally when using `throwIllegalValue` I've seen the `istanbul ignore` comment above the `default` to make sure that it doesn't detract from test coverage. Worth looking into

Reply:
Ah yes thanks for the reminder! Ignored this default block and a few others I'd recently added and verified that that did in fact bring our coverage percentages up. Also looks like we hadn't bumped up our test coverage thresholds in a while so did that too 📈